### DISCUSSION TITLE
migration from perforce to github

### DISCUSSION THREAD

**Discussion by me948** [community/community#30757] [Other Feature Feedback, Questions, & Ideas]
I am able to clone the repository from perforce using git p4.py clone command .But while pushing  the repository I'm getting the below error. My repository has 5 GB of data. Can we push the repository (with 5 GB of data and large files) into GitHub without using git LFS?

 git push origin master
Enumerating objects: 5671, done.
Counting objects: 100% (5671/5671), done.
Delta compression using up to 2 threads
Compressing objects: 100% (2626/2626), done.
remote: fatal: pack exceeds maximum allowed size (2.00 GiB)
fatal: sha1 file '<stdout>' write error: Broken pipe
error: remote unpack failed: index-pack abnormal exit

**Comment by airtower-luna** [comment]
Possibly, if you don't have any files larger than 50 MB (that's a hard limit). The [documentation mentions](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github):

> GitHub blocks pushes that exceed 100 MB.

So if the individual files are small enough you might be able to push in chunks smaller than 100 MB.

However 5GB is extremely big for a Git repository. Unless it's one huge codebase (like the Linux kernel) you should probably consider splitting it into multiple repositories. I unfortunately had to deal with Perforce for work for a while, and it seems to encourage stuffing everything into one "depot". :sweat_smile:
